libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1317,10 +1350,34 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::SUB;
 		break;
 	case Token::Mul:
-		m_context << Instruction::MUL;
+		if (c_isFractional)
+		{
+			//multiply, then shift right...this is your fraction.
```

Comment:
so here's what I'm doing. Essentially, whenever you multiply a fixed point number, you have to take into account the format of what it's set up, and add them together. So if I have a fixed128x128 and I multiply another fixed128x128, then I'm going to get a fixed256x256, which obviously doesn't fit. To correct this, what we do is we simply multiply, grab the fractional part from the overflow, and shift it back down and truncate it off. After that, we then shift down the integer portion so that it acts more like a regular integer, and multiply that by the original value, and then we add the fractional and integer results together to get our final result.